src/provider-registry/index.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * Provider Registry - Singleton for managing AI providers
+ *
+ * This module implements a singleton registry that allows dynamic registration
+ * of AI providers at runtime, while maintaining compatibility with the existing
+ * static PROVIDERS object in ai-services-unified.js.
+ */
+
+import { log } from '../../scripts/modules/utils.js';
+
+// Singleton instance
+let instance = null;
+
+/**
+ * Provider Registry class - Manages dynamic provider registration
+ */
+class ProviderRegistry {
+	constructor() {
+		// Private provider map
+		this._providers = new Map();
+
+		// Flag to track initialization
+		this._initialized = false;
+
+		log('debug', 'Provider Registry created');
+	}
+
+	/**
+	 * Get the singleton instance
+	 * @returns {ProviderRegistry} The singleton instance
+	 */
+	static getInstance() {
+		if (!instance) {
+			instance = new ProviderRegistry();
+		}
+		return instance;
+	}
+
+	/**
+	 * Initialize the registry
+	 * @returns {ProviderRegistry} The singleton instance
+	 */
+	initialize() {
+		if (this._initialized) {
+			log('debug', 'Provider Registry already initialized');
+			return this;
+		}
+
+		log('info', 'Initializing Provider Registry');
```

Comment:
make it a debug not an info
```suggestion
		log('debug', 'Initializing Provider Registry');
```